src/rlx_app_info.erl

```diff
@@ -56,146 +54,104 @@
-                             vsn=ParsedVsn,
-                             dir=Dir,
-                             active_deps=ActiveDeps,
-                             library_deps=LibraryDeps,
-                             link=Link}}
-    end.
-
--spec name(t()) -> atom().
-name(#app_info_t{name=Name}) ->
+    {ok, rlx_app:new(AppName, Vsn, Dir, ActiveDeps, LibraryDeps, Link)}.
+
+-spec name(rlx_app:t()) -> atom().
+name(#{name := Name}) ->
     Name.
 
--spec name(t(), atom()) -> t().
-name(AppInfo=#app_info_t{}, AppName)
+-spec name(rlx_app:t(), atom()) -> rlx_app:t().
+name(AppInfo, AppName)
   when erlang:is_atom(AppName) ->
-    AppInfo#app_info_t{name=AppName}.
+    AppInfo#{name => AppName}.
 
--spec vsn(t()) -> ec_semver:semver().
-vsn(#app_info_t{vsn=Vsn}) ->
+-spec vsn(rlx_app:t()) -> ec_semver:semver().
+vsn(#{vsn := Vsn}) ->
     Vsn.
 
--spec original_vsn(t()) -> string().
-original_vsn(#app_info_t{original_vsn=Vsn}) ->
-    Vsn.
+-spec vsn(rlx_app:t(), string()) -> {ok, rlx_app:t()} | relx:error().
+vsn(AppInfo, Vsn)
+  when erlang:is_list(Vsn) ->
+    {ok, AppInfo#{vsn => Vsn}}.
 
--spec vsn_as_string(t()) -> string().
-vsn_as_string(#app_info_t{vsn=Vsn}) ->
-    erlang:binary_to_list(erlang:iolist_to_binary(ec_semver:format(Vsn))).
-
--spec vsn(t(), string()) -> {ok, t()} | relx:error().
-vsn(AppInfo=#app_info_t{name=AppName}, AppVsn)
-  when erlang:is_list(AppVsn) ->
-    case parse_version(AppVsn) of
-        {fail, _} ->
-            ?RLX_ERROR({vsn_parse, AppName});
-        ParsedVsn ->
-            {ok, AppInfo#app_info_t{vsn=ParsedVsn}}
-    end.
-
--spec dir(t()) -> binary().
-dir(#app_info_t{dir=Dir}) ->
+-spec dir(rlx_app:t()) -> binary().
+dir(#{dir := Dir}) ->
     Dir.
--spec dir(t(), binary()) -> t().
-dir(AppInfo=#app_info_t{}, Dir) ->
-    AppInfo#app_info_t{dir=Dir}.
 
--spec active_deps(t()) -> [atom()].
-active_deps(#app_info_t{active_deps=Deps}) ->
+-spec dir(rlx_app:t(), binary()) -> rlx_app:t().
+dir(AppInfo, Dir) ->
+    AppInfo#{dir => Dir}.
+
+-spec active_deps(rlx_app:t()) -> [atom()].
+active_deps(#{applications := Deps}) ->
     Deps.
--spec active_deps(t(), [atom()]) -> t().
-active_deps(AppInfo=#app_info_t{}, ActiveDeps)
+
+-spec active_deps(rlx_app:t(), [atom()]) -> rlx_app:t().
+active_deps(AppInfo, ActiveDeps)
   when erlang:is_list(ActiveDeps) ->
-    AppInfo#app_info_t{active_deps=ActiveDeps}.
+    AppInfo#{applications => ActiveDeps}.
 
--spec library_deps(t()) -> [atom()].
-library_deps(#app_info_t{library_deps=Deps}) ->
+-spec library_deps(rlx_app:t()) -> [atom()].
+library_deps(#{included_applications := Deps}) ->
     Deps.
 
--spec library_deps(t(), [atom()]) -> t().
-library_deps(AppInfo=#app_info_t{}, LibraryDeps)
+-spec library_deps(rlx_app:t(), [atom()]) -> rlx_app:t().
+library_deps(AppInfo, LibraryDeps)
   when erlang:is_list(LibraryDeps) ->
-    AppInfo#app_info_t{library_deps=LibraryDeps}.
+    AppInfo#{included_applications => LibraryDeps}.
```

Comment:
odd choice. I guess library_deps maps to apps that don't get started for now?

Reply:
Na, those literally are the list of `included_applications`. The use of the term `LibraryDeps` is what is out of place and will be fixed as `rlx_app_info` is replaced by the `rlx_app` module.